mne/decoding/xdawn.py

```diff
@@ -0,0 +1,177 @@
+"""Contains refactored Xdawn class."""
+
+# Authors: Alexandre Barachant <alexandre.barachant@gmail.com>
+#          Asish Panda <asishrocks95@gmail.com>
+#
+# License: BSD (3-clause)
+
+import numpy as np
+
+from ..preprocessing.xdawn import _Xdawn
+from . import TransformerMixin
+
+
+class XdawnTransformer(TransformerMixin, _Xdawn):
+
+    """Implementation of the Xdawn Algorithm.
+
+    Xdawn is a spatial filtering method designed to improve the signal
+    to signal + noise ratio (SSNR) of the ERP responses. Xdawn was originally
+    designed for P300 evoked potential by enhancing the target response with
+    respect to the non-target response. This implementation is a light version
+    of the original version and follows scikit-learn API strictly.
+
+    .. note:: This does not handle overlapping events. Use original
+              preprocessing.xdawn instead.
+
+    Parameters
+    ----------
+    n_components : int (default 2)
```

Comment:
@dengemann I did not change it from the original `Xdawn`. [Link](https://github.com/mne-tools/mne-python/blob/b6bab37a6154aa21c629ad19ac62f49f08f249bb/mne/preprocessing/xdawn.py#L168)